projects/igniteui-angular/src/lib/grids/toolbar/grid-toolbar-advanced-filtering.component.ts

```diff
@@ -50,6 +50,26 @@ export class IgxGridToolbarAdvancedFilteringComponent implements AfterViewInit {
         this.grid?.advancedFilteringExpressionsTreeChange.subscribe(filteringTree => {
             this.numberOfColumns = this.extractUniqueFieldNamesFromFilterTree(filteringTree).length;
         });
+        this.grid?.filteringExpressionsTreeChange.subscribe(() => {
```

Comment:
Advanced filtering is using advancedFilteringExpressionsTree collection why is necessary to listen on filteringExpressionsTreeChange here?!

Reply:
I listen for a change in the filtering expression tree `filteringExpressionsTreeChange` because when we change the data by default the filters are cleared and `filteringExpressionsTreeChange` is executed while `advancedFilteringExpressionsTreeChange` is not executed, and it is executed only if we clear the filters from the **'Clear Filters'** button in the advanced filtering dialog.

That's why in this scenario when the data changes, apart from this PR fixing the issue about the error that appears in the console when opening the advance filter dialog, I also handled that to clear the advance filtering expression tree and to reset the number that shows how many filters are attached. 

That's why when changing the data and clearing the filters, I listen on `filteringExpressionsTreeChange` event to see if it is necessary to clear the advanced filter and also to clear the number, because `advancedFilteringExpressionsTreeChange` is not executed in this case.

**Not sure if this should be handled in this PR or if a new issue should be logged for this case?!**